DB/Item.lua

```diff
@@ -165,4 +165,83 @@ function Item:HasZone(entry)
 	return false
 end
 
+function Item:IsCollectionItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.COLLECTION
+end
+
+function Item:IsUsingCollectionProperties(entry)
```

Comment:
Is this type of naming an accurate description? IIRC in the first version I named those methods ``IsValidXYZ`` because the function would return whether the sum of the item's properties amount to a valid item of the given type. ``IsUsingXyzProperties`` sounds like there's some sort of provider for Xyz properties?

Not sure what to make of this, perhaps we can find a better description for these types of validation methods? I thought of them as similar to unit tests, though not explicitly labeled as such, but I figure either way we should keep names consistent and agree on a convention here :)

Reply:
Well, you did name the first one `Item:IsUsingHolidayProperties(entry)`, so I simply copied the format you used on that one.

I overlooked the `Item:IsValidHolidayItem(entry)` function that's already implemented for HolidayProperties, so I took a slightly different approach. If I am to follow the structure used on holiday items, then there should be 3 functions
Item:IsXYZItem(entry) - Just to check the entry is specific category/method/something
Item:IsValidXYZItem(entry) - Check that all interactions are set, and if needed, have the right values
Item:IsUsingXYZProperties(entry) - Is using fields tied to this case. No other entries should use these properties

If you think this looks right, then I'll simply add the 'IsValidXYZItem' onces and update the schema accordingly

Reply:
Obvious disclaimer: My names aren't usually great :joy: 

If I remember correctly, holiday items were a bit of a weird case. I agree with haxing ``IsXyzItem()`` as a shorthand for the basic category check, and ``IsValidXyz()`` can take care of the validation.

Not sure if the ``IsUsingXyzProperties()`` is really needed though? Maybe we can just remove it, as it would be part of the validation logic to check that a field is of the appropriate type, which implies it can't be ``nil`` (since ``typeof(nil) == "nil``.

This means there would only be two functions/checks for each item type, which makes things easier. I'm a big fan of making things easier, but perhaps there's a better way?

Reply:
Sure, I don't really mind taking away the `IsUsingXyzProperties()` ones. It's function is basically to warn if someone stuffed things like 'zones' on BOSS items, or 'raceid' on FISHING items, but it's just a warning at this point either way.

Reply:
You don't have to take them out, obviously :) Having more validation isn't a bad thing, and if things get too unwieldy we can just move them to a different file/restructure it.

Reply:
Rewrote most of the validations, so they generally only use Item:IsXYZItem(entry) and Item:IsValidXYZItem(entry)